src/app/models/propTypes/general/index.js

```diff
@@ -1,12 +1,49 @@
-import { shape, string, oneOf, arrayOf, oneOfType } from 'prop-types';
+import {
+  shape,
+  string,
+  oneOf,
+  arrayOf,
+  oneOfType,
+  checkPropTypes,
+} from 'prop-types';
+import requiredPropCheck from './requiredPropCheck';
+
+export const blockObjectOfSpecificTypeAndModel = (type, model) => ({
+  blockId: string.isRequired,
+  type: oneOf([type]).isRequired,
```

Comment:
Why is this `oneOf([type]).isRequired`? Should it not be `string.isRequired`, since `type` can be `'rawImage'`, etc.